diff_cover/git_diff.py

```diff
@@ -62,7 +62,7 @@ def diff_committed(self, compare_branch="origin/main"):
         Raises a `GitDiffError` if `git diff` outputs anything
         to stderr.
         """
-        diff_range = "{branch}{notation}HEAD".format(
+        diff_range = compare_branch if self.range_notation in compare_branch else "{branch}{notation}HEAD".format(
```

Comment:
So I think you are sneaking in a bit of a hack.

However, it seems like a useful hack. It might be better to just make the `HEAD` part configurable explicitly rather than doing this trick.

But if you don't want to do that I think im willing to let this in... but you need to at least include a comment explaining this above so I remember why this exists when I see this code again